src/run/headers.ts

```diff
@@ -242,14 +250,27 @@ export const setCacheControlHeaders = (
       .log('NetlifyHeadersHandler.trailingSlashRedirect')
   }
 
-  if (status === 404 && request.url.endsWith('.php')) {
-    // temporary CDN Cache Control handling for bot probes on PHP files
-    // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
-    headers.set('cache-control', 'public, max-age=0, must-revalidate')
-    headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+  const cacheControl = headers.get('cache-control')
+  if (status === 404) {
+    if (request.url.endsWith('.php')) {
+      // temporary CDN Cache Control handling for bot probes on PHP files
+      // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
+      headers.set('cache-control', 'public, max-age=0, must-revalidate')
+      headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+      return
+    }
+
+    if (
+      ['GET', 'HEAD'].includes(request.method) &&
+      !headers.has('cdn-cache-control') &&
+      !headers.has('netlify-cdn-cache-control')
+    ) {
```

Comment:
In addition to removing `!cacheControl || cacheControl.includes('no-store')` from the condition, I found I had to also remove `typeof requestContext.pageHandlerRevalidate !== 'undefined'` because 404s without `getStaticProps` do not go through the cache handler so `pageHandlerRevalidate` is `undefined` in those cases. Was it there as a precaution, or was there a specific case you were guarding against? @pieh

Reply:
> Was it there as a precaution, or was there a specific case you were guarding against? @pieh

Precaution mainly (same like previous `cache-control`), but I think the change you made make sense overall - it does widen scenarios of 404 caching where it possibly could not be intentional, but at the same time it catch more cases of 404 not being cached today